feat: 사내망 Harbor 레지스트리 분기 처리 및 스크립트 개선

suhyun-kim415 opened the pull request:
- 모든 애플리케이션의 이미지 pull 경로에 Harbor 분기 처리 추가
- internalHarbor.enabled가 true일 때 사내망 Harbor 사용
- 외부망일 때는 원본 레지스트리(docker.io, quay.io 등) 사용

주요 변경사항:
- applications/*: 모든 애플리케이션 이미지에 Harbor 분기 처리 추가
  * astrago: ASTRAGO_PRIVATE-REGISTRY-URL 환경 변수 분기 처리
  * prometheus: Prometheus 스택 이미지 분기 처리
  * harbor, ingress-nginx, keycloak, flux, gpu-operator, csi-driver-nfs, mpi-operator: 각각의 이미지 registry 분기 처리
- environments/prod/values.yaml: internalHarbor 설정 추가
- kubespray/deploy-kubespray.sh: macOS realpath 호환성 개선
- helmfile.yaml, deploy_astrago.sh: Harbor 관련 설정 반영

Harbor 프로젝트 매핑:
- docker.io, ghcr.io → /dockerhub
- quay.io → /quay
- registry.k8s.io → /k8s
- nvcr.io → /nvcr

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Per-component conditional image registry selection (internal Harbor, offline, or public) applied across many charts.
  * Added network-operator release and consolidated ingress-nginx deployment.

* **Chores**
  * Updated production external IP, NFS server, and multiple image tags; enabled internal Harbor endpoint.
  * Improved deployment scripts for macOS/Linux (Python handling) and minor deploy script PATH fix.

* **Refactor**
  * Replaced global registry with per-component registry resolution.

* **Bug Fixes**
  * Removed a controller toleration entry and aligned ingress images with conditional registries.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

suhyun-kim415 committed:
feat: 사내망 Harbor 레지스트리 분기 처리 및 스크립트 개선

- 모든 애플리케이션의 이미지 pull 경로에 Harbor 분기 처리 추가
- internalHarbor.enabled가 true일 때 사내망 Harbor 사용
- 외부망일 때는 원본 레지스트리(docker.io, quay.io 등) 사용

주요 변경사항:
- applications/*: 모든 애플리케이션 이미지에 Harbor 분기 처리 추가
  * astrago: ASTRAGO_PRIVATE-REGISTRY-URL 환경 변수 분기 처리
  * prometheus: Prometheus 스택 이미지 분기 처리
  * harbor, ingress-nginx, keycloak, flux, gpu-operator, csi-driver-nfs, mpi-operator: 각각의 이미지 registry 분기 처리
- environments/prod/values.yaml: internalHarbor 설정 추가
- kubespray/deploy-kubespray.sh: macOS realpath 호환성 개선
- helmfile.yaml, deploy_astrago.sh: Harbor 관련 설정 반영

Harbor 프로젝트 매핑:
- docker.io, ghcr.io → /dockerhub
- quay.io → /quay
- registry.k8s.io → /k8s
- nvcr.io → /nvcr

lannex commented:
반영을 위해 pr 닫고 다시 열어볼게요.

lannex commented:
@coderabbitai summary

lannex commented:
코드래빗으로 다시 리뷰 태우는 방법을 해보는데 설명처럼 제대로 동작하지 않네요. 
그래서 이번 pr에선 코드래빗 코멘트를 무시해도 되게 처리했습니다.
다음 pr에선 수정된 코드래빗 리뷰가 동작할거에요.

suhyun-kim415 committed:
Fix mariadb image

mariadb 오프라인 설정 추가
mariadb:
  image:
    {{- else if .Values.offline.registry }}
    registry: "{{ .Values.offline.registry }}"

suhyun-kim415 committed:
deploy_astrago.sh - 잠재이슈 제거

잠재적 이슈 제거 (경로 불일치로 인해 배포가 실패)

suhyun-kim415 committed:
Update deploy-kubespray.sh

주석 불일치 문구 제거,    ./setup-py.sh || exit 1 방어코드

suhyun-kim415 committed:
Update deploy-kubespray.sh

방어로직 - cd "$CURRENT_DIR" || exit

suhyun-kim415 commented:
이슈 취소로 인한 PR close
